### DISCUSSION TITLE
BOUNTY: Decode MSN stock IDs to ticker symbols

### DISCUSSION THREAD

**Discussion by rcmaehl** [rcmaehl/MSEdgeRedirect#339] [Help Bounties]
Currently, I am unable to come up with a solution to redirect stocks. This is because MSN stock IDs make no sense.

E.g. Tesla is a24kar or https://www.msn.com/en-us/money/watchlist?id=a24kar

Provide a list of stock IDs to ticket symbols so they can be added to MSEdgeRedirect

BOUNTY: $50 USD

**Comment by whamer100** [comment]
One solution that could be done is to find MSN stock IDs is to use this one endpoint i found: `https://services.bingapis.com/contentservices-finance.csautosuggest/api/v1/Query?query=TSLA&market=en-us&count=1` changing the query to a valid stock symbol results in a response where inside the json string is a key called `SecId` which is the MSN stock ID. This api can be queried on a known list of stock symbols to build a list. (that I do not have a list of at hand)

**Reply by whamer100** [reply]
Just wrote a quick script that's iterating through a pretty sizeable list of known stocks from a ftp from a nasdaq website. Will post the resulting list of stock IDs to ticket symbols once this finishes running.

**Reply by whamer100** [reply]
The source csv document I'm using appears to have ~11.2k symbols defined, however not all of them appear to get resolved through the api endpoint I'm using. Fortunately, most of them appear to get resolved. Script still taking its time...

**Reply by whamer100** [reply]
Ok cool it finished, here's a list of stocks mapped to their msn stock ID counterpart: [stocks.txt](https://github.com/rcmaehl/MSEdgeRedirect/files/13376427/stocks.txt)

10.1k of the stocks were successfully mapped using the api endpoint listed above.

**Reply by rcmaehl** [reply]
Hi @whamer100, do you have a paypal or something else I can send the bounty to?

**Reply by whamer100** [reply]
@rcmaehl yes i do (i actually just set one up as i realized i didnt actually do that yet) whamer100.forpaypal@gmail.com

**Reply by whamer100** [reply]
i havent really used paypal this way before so let me know if its not working

**Reply by rcmaehl** [reply]
For some reason it will not let me send it fee-less "friends and family". I've sent $104.12 for both bounties which should be 
100 USD after 3.49% + 0.49 USD fees paypal takes out. If it doesn't, please let me know.

**Reply by whamer100** [reply]
Yes i did receive $100. Currently just waiting for PayPal to release its hold on the transaction.
